### PR TITLE
feat: Add Iot client

### PR DESCRIPTION
**Pull request by tyx**
Hello !

Here is a new client to allow us to do some Iot stuff.

To start we mainly need to create and list things.

Hope nothing is missing.

### CODE REVIEW

**Comment by tyx** [comment]
I just realized that iot is only supported in pro version for localstack. How am I supposed to do integration test ?

**Comment by jderusse** [comment]
> I just realized that iot is only supported in pro version for localstack. How am I supposed to do integration test ?

In that situation, we don't provide any integration test.

**Comment by tyx** [comment]
So I just need to remove the integration folder ?

**Comment by tyx** [comment]
I also had trouble with psalm but with another Client (Cognito) so unrelated to this PR I guess

File: src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php [hunk 1]
```diff
@@ -111,7 +111,7 @@ public function generateForPath(StructureShape $shape, string $path, string $out
 
     protected function parseResponseTimestamp(Shape $shape, string $input, bool $required): string
     {
-        $body = 'new \DateTimeImmutable((string) INPUT)';
+        $body = 'new \DateTimeImmutable((string) (is_numeric(INPUT) ? "@{INPUT}" : INPUT))';
```
Comment (tyx) [inline src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php:114]: Allow to handle timestamp unix with microseconds

**Comment by tyx** [comment]
I reverted my change in generated code.

I succeed in doing one change in COdeGenerator to handle my timestamp with microseconds but the others PHPStan/Psalm failure looks more weird.

**Comment by jderusse** [comment]
> So I just need to remove the integration folder ?

adds a `markTestIncomplete` on top of the test (You'll find several occurrences in other services)


> But I just see that php 7.2 does not support `DateTime::createFromFormat` disappointed

Hmm... I believe this should work. See https://3v4l.org/DlCbH#v7.2.34

File: src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php [hunk 2]
```diff
@@ -111,7 +111,7 @@ public function generateForPath(StructureShape $shape, string $path, string $out
 
     protected function parseResponseTimestamp(Shape $shape, string $input, bool $required): string
     {
-        $body = 'new \DateTimeImmutable((string) INPUT)';
+        $body = 'is_numeric(INPUT) ? \Datetime::createFromFormat("U.u", sprintf("%.6F", INPUT)) : new \DateTimeImmutable((string) INPUT)';
```
Comment (jderusse) [inline src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php:114]: What is the prupose of the check on `is_numeric` do you have example of non numeric data?

Comment (tyx) [reply src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php:114]: Well I did not want to break anything on other Client as I don't know how other timestamp are returned by aws !

But if we are sure we have real timestamp everywhere I can remove it no problem !

Comment (jderusse) [reply src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php:114]: I think the `sprintf` thing would fix the input. https://3v4l.org/lXAQc#v7.2.34

Comment (jderusse) [reply src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php:114]: hmmm... looks like this is already what we are doing in `JsonRpcParser`

**Comment by tyx** [comment]
Thanks @jderusse I succeeded in fixing the whole CI

File: src/Core/src/AwsClientFactory.php [hunk 3]
```diff
@@ -292,6 +293,19 @@ public function iam(): IamClient
         return $this->serviceCache[__METHOD__];
     }
 
+    public function iot(): IotClient
```
Comment (jderusse) [inline src/Core/src/AwsClientFactory.php:296]: Please add an entry in CHANGELOG

File: src/Integration/Symfony/Bundle/src/DependencyInjection/AwsPackagesProvider.php [hunk 4]
```diff
@@ -73,6 +73,10 @@ public static function getAllServices(): array
                 'class' => \AsyncAws\Iam\IamClient::class,
                 'package' => 'async-aws/iam',
             ],
+            'iot' => [
```
Comment (jderusse) [inline src/Integration/Symfony/Bundle/src/DependencyInjection/AwsPackagesProvider.php:76]: Please add an entry in CHANGELOG

**Review by jderusse — COMMENTED** [review]
Thank you for this contribution. 

I've just 2 minor comments about adding references in related CHANGELOG files

File: src/Core/src/AwsClientFactory.php (hunk 3 above)
Comment (tyx) [reply src/Core/src/AwsClientFactory.php:296]: done

**Pull request merged**